From: coolsnowwolf <coolsnowwolf@gmail.com>
Subject: [PATCH] nss-drv: consume external compiler flags

Linux 6.18 no longer consumes EXTRA_CFLAGS in the top-level Kbuild.
Add the package-provided include paths to the module flags directly.

Signed-off-by: coolsnowwolf <coolsnowwolf@gmail.com>
---
--- a/Makefile
+++ b/Makefile
@@ -599,6 +599,7 @@
 ccflags-y += -I$(obj)/nss_data_plane/hal/include
 ccflags-y += -DNSS_PM_DEBUG_LEVEL=0 -DNSS_SKB_REUSE_SUPPORT=1
 ccflags-y += -Wall -Werror -Wno-missing-declarations -Wno-missing-prototypes -Wno-empty-body
+ccflags-y += $(EXTRA_CFLAGS)
 
 KERNELVERSION := $(word 1, $(subst ., ,$(KERNELVERSION))).$(word 2, $(subst ., ,$(KERNELVERSION)))
 
--- a/nss_init.c
+++ b/nss_init.c
@@ -143,6 +143,7 @@
 	{ .compatible = "qcom,nss" },
 	{ .compatible = "qcom,nss0" },
 	{ .compatible = "qcom,nss1" },
+	{ }
 };
 MODULE_DEVICE_TABLE(of, nss_dt_ids);
 #endif
